packages/react/core/README.md

```diff
@@ -56,6 +56,98 @@ function App() {
 
+### `useConnect(reactWallet)`
+
+Given a `ReactWallet`, this hook returns a function you can call to ask the wallet to authorize the current domain to use new accounts. Calling the returned method will return a promise for the array of account authorized by the wallet.
+
+```tsx
+function ConnectButton({ onAccountsConnected, wallet }) {
+    const [isConnecting, connect] = useConnect(wallet);
+    return (
+        <button
+            disabled={isConnecting}
+            onClick={() => {
+                connect.then(onAccountsConnected);
+            }}
+        >
+            Connect Wallet
+        </button>
+    );
+}
+```
+
+Be aware that calls to `connect` operate on a single promise that is global to the entire app. If you call `connect` multiple times or from multiple places in your app, the exact same promise will be vended to all callers. This reflects the fact that a wallet can only field one authorization request at a time.
+
+### `useDisconnect(reactWallet)`
+
+Given a `ReactWallet`, this hook returns a function you can call to ask the wallet to deauthorize accounts authorized for the current domain, or at least to remove them from the wallet's `accounts` array for the time being, depending on the wallet's implementation of `standard:disconnect`. Call the returned method will return a promise that resolves once the disconnection is complete.
+
+```tsx
+function DisconnectButton({ onDisconnected, wallet }) {
+    const [isDisconnecting, disconnect] = useDisconnect(wallet);
+    return (
+        <button
+            disabled={isDisconnecting}
+            onClick={() => {
+                disconnect.then(onDisconnected);
+            }}
+        >
+            Disconnect
+        </button>
+    );
+}
+```
+
+### `useSignMessage(reactWalletAccount)`
+
+Given a `ReactWalletAccount`, this hook returns a function you can call to sign a byte array using that account's private key.
+
+```tsx
+function MessageSignerPanel({ reactWalletAccount }) {
+    const signMessage = useSignMessage(reactWalletAccount);
+    const textareaRef = createRef();
+    const handleClick = useCallback(() => {
+        const message = textareaRef.current.value;
+        const messageBytes = new TextEncoder().encode(message);
+        signMessage(messageBytes).then(({ signature, signedMessage }) => {
+            console.log('You signed a message!', message, new TextDecoder('utf8').decode(signedMessage), signature);
+        });
+    }, [signMessage]);
+    return (
+        <section>
+            <textarea ref={textareaRef}></textarea>
+            <button onClick={handleClick}>Sign Message</button>
+        </section>
+    );
+}
+```
+
+### `useSignAndSendTransaction(reactWalletAccount, chain)`
```

Comment:
What do you think about `useSignTransaction` here?

I think in Solana the direction of travel toward sign + send is clear, but I'm not sure it makes sense to reflect that in `wallet-standard`. And we do have an `experimental:signTransaction` feature here.

I also think different wallets and apps are going to move at different rates, and if we force no `signTransaction` here then it's a reason for Solana apps not to adopt this yet, until they've moved to sign + send. And this enables adopting the new web3js, so it'd be a shame to give apps blockers to adoption.

IMO it makes sense to provide the hook, and wallets can control its availability and behaviour with features and their own implementation.

Reply:
Oh yeah, for sure I can add it. I just forgot.